bip-taro.mediawiki

```diff
@@ -392,18 +392,16 @@ derived using the first input outpoint as a sort of chain randomness to ensure
 uniqueness within the system as well as the output index and asset type to
 prevent collisions.
 The <code>asset_group_key</code> is derived as follows:
-* <code>asset_group_key</code> = <code>asset_key_internal +
-sha256(asset_key_internal || genesis_outpoint || output_index || asset_type)
+* <code>asset_group_key</code> = <code>TapTweak(asset_internal_group_key,
+ root)
 </code>
 
 where:
-* <code>asset_key_internal</code> is a 32-byte public key as defined by
+* <code>asset_internal_group_key = singleTweak(asset_raw_key, asset_id)</code> 
```

Comment:
So we can either prescribe that the asset iD is verified, or "unroll" the computation like we did above. Leaning towards the latter. 

There's another related change here: rn we include all the "asset genesis" (so the input to the asset ID) within each leaf TLV. Instead we can just have the `asset_id` there (single field), and move the genesis reveal to the genesis proof like we've done with the asset meta reveal.

Reply:
Genesis reveal moved to proof file.